File: Array/1094.Car Pooling/Solution.py
class Solution:
    def carPooling(self, trips: List[List[int]], capacity: int) -> bool:
        q = [0] * 1010
        for n, u, v in trips:
            q[u] += n
            q[v] -= n
        s = 0
        for v in q:
            s += v
            if s > capacity:
                return False
        return True

File: Array/1094.Car Pooling/Solution.java
class Solution {
    public boolean carPooling(int[][] trips, int capacity) {
        int[] q = new int[1010];
        for (int[] t : trips) {
            int n = t[0], u = t[1], v = t[2];
            q[u] += n;
            q[v] -= n;
        }
        int s = 0;
        for (int v : q) {
            s += v;
            if (s > capacity) {
                return false;
            }
        }
        return true;
    }
}